### DISCUSSION TITLE
How to set default terminal to be powershell?

### DISCUSSION THREAD

**Discussion by deostroll** [wezterm/wezterm#809] [Q&A]
How to do the following:

1. set the default terminal to be powershell. 
2. When spawning a new tab I want it to be at the same location as the previous tab was...

Any ideas?

**Comment by wez** [comment]
Take a look at:
https://wezfurlong.org/wezterm/config/launch.html#changing-the-default-program
You can set it to `pwsh.exe`

And for the location, I haven't tried this myself, but I think you'll need to configure powershell with something this so that it sets "OSC 7" to tell the terminal where it is:
https://gist.github.com/skyline75489/480d036db8ae9069b7009377e6eebb79#gistcomment-3460396

There's a bit more information about that aspect of shell integration in these docs: https://wezfurlong.org/wezterm/shell-integration.html

Please share what you find; I think this would make a great addition to the docs!

**Reply by deostroll** [reply]
Thanks. I was able to make it powershell, but I couldn't quite understand the OSC - I mean how to go about it.

**Reply by wez** [reply]
I don't use powershell, so I don't know either :-p
The idea though is that you need to configure powershell to output the OSC sequence so that it can tell wezterm what the current working directory is.  Once that is set, opening new panes/tabs will use that working directory.

**Reply by wez** [reply]
I took a quick look today; you need to edit or create your powershell profile and define a prompt function; the profile path is pre-defined and stored in `$PROFILE` in powershell, which is a path like this:

```
PS C:\Users\wez> echo $Profile
C:\Users\wez\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1
```

Create and edit that file and define a prompt function like this:

```powershell
function prompt {
    $p = $executionContext.SessionState.Path.CurrentLocation
    $osc7 = ""
    if ($p.Provider.Name -eq "FileSystem") {
        $ansi_escape = [char]27
        $provider_path = $p.ProviderPath -Replace "\\", "/"
        $osc7 = "$ansi_escape]7;file://${env:COMPUTERNAME}/${provider_path}${ansi_escape}\"
    }
    "${osc7}PS $p$('>' * ($nestedPromptLevel + 1)) ";
}
```

that emits OSC 7 with the current directory whenever the prompt is printed.
